src/main/java/teammates/logic/api/Logic.java

```diff
@@ -2622,4 +2622,14 @@ public MimeMessage emailErrorReport(String path, String params, Throwable error)
     @SuppressWarnings("unused")
     private void ____helper_methods________________________________________() {
     }
+    
+    public void extractActiveAndArchivedCourses(List<CourseDetailsBundle> courseBundles, 
```

Comment:
The way this is used, you can alternatively make the method return a map with your own defined keys and those three lists as values.